ts/types/errors.ts

```diff
@@ -0,0 +1,20 @@
+export class StorexError extends Error {
+    constructor(msg: string) {
+        super(msg)
+        Object.setPrototypeOf(this, new.target.prototype)
```

Comment:
What does this do? Could use a comment  :)

Reply:
It's to combat this TS issue that was introduced a while back when extending from a bunch of in-built constructors, like `Error`:
https://github.com/Microsoft/TypeScript-wiki/blob/master/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work

Added a comment.